packages/react-magma-dom/src/components/DateTimePicker/index.tsx

```diff
@@ -0,0 +1,267 @@
+   */
+  onDone?: (event: React.SyntheticEvent) => void;
+}
+
+const DoneButtonWrapper = styled.div<{ isInverse?: boolean }>`
+  background-color: ${props =>
+    props.isInverse
+      ? props.theme.colors.primary600
+      : props.theme.colors.neutral200};
+  padding: 16px;
+  margin: 0 -16px -8px;
+  display: flex;
+  justify-content: flex-end;
+`;
+
+export const DateTimePicker = React.forwardRef<
+  HTMLInputElement,
+  DateTimePickerProps
+>((props, forwardedRef) => {
+  const {
+    onTimeChange,
+    onDone,
+    placeholder,
+    value,
+    defaultValue,
+    labelText,
+    timePickerLabelText,
+    buttonLabelText,
+    ...other
+  } = props;
+  const i18n = React.useContext(I18nContext);
+
+  const getTimeFromDate = (date: Date | number) => {
+    if (!date) return '';
+
+    const dateObj = new Date(date);
+    return dateObj.toLocaleTimeString('en-US', {
+      hour12: true,
+      hour: 'numeric',
+      minute: '2-digit',
+    });
+  };
+
+  const initialTime = getTimeFromDate(value ?? defaultValue);
+
+  const theme = React.useContext(ThemeContext);
+  const datePickerApiRef = React.useRef(null);
+  const previousTime = React.useRef<string>('');
+  const [additionalInputContent, setAdditionalInputContent] = React.useState(
+    initialTime || ''
+  );
+
+  const handleDoneClick = (event: React.SyntheticEvent) => {
+    datePickerApiRef.current?.closeDatePickerManually();
+
+    onDone && onDone(event);
+  };
+
+  const handleClear = () => {
+    setAdditionalInputContent('');
+    previousTime.current = '';
+  };
+
+  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    if (!value) {
+      setAdditionalInputContent('');
+      previousTime.current = '';
+      return;
+    }
+
+    const timeMatch = value.match(/\b(\d{1,2}:\d{2}\s?[AP]M)\b/i);
+
+    if (timeMatch) {
+      const timeValue = timeMatch[1];
+      setAdditionalInputContent(timeValue);
+      previousTime.current = timeValue;
+    } else {
+      const cuttedValue = value.split(' ');
+      if (cuttedValue[0] && cuttedValue[0] !== '') {
+        setAdditionalInputContent(previousTime.current);
+      } else {
+        setAdditionalInputContent('');
+        previousTime.current = '';
+      }
+    }
+  };
+
+  const onTimeHandleChange = (value: string) => {
+    setAdditionalInputContent(value);
+    previousTime.current = value;
+
+    onTimeChange && onTimeChange(value);
+  };
+
+  React.useEffect(() => {
+    if (!value && !defaultValue) {
+      handleClear();
+    } else if (value) {
+      const newTime = getTimeFromDate(value);
+
+      setAdditionalInputContent(newTime);
+      previousTime.current = newTime;
+    }
+  }, [value, defaultValue]);
+
+  const dateFormat = i18n.dateFormat;
+  const updatedPlaceholder = placeholder
+    ? placeholder
+    : `${dateFormat.toLowerCase()} hh:mm AM`;
+
+  return (
+    <DatePicker
+      labelText={labelText ?? 'Pick a date and time'}
```

Comment:
@orion-cengage can you confirm you are okay with this default? or do you have a different opinion?